(WIP): CNV Release Notes 2.2 Working Draft in Progress

bgaydosrh opened the pull request:
This PR is intended for the collection of CNV 2.2 release notes. I removed the new and changed features from 2.1 but left the known issues in case we want to move some of them to resolved.

Here is the link to the Google Doc which lists status of Known Issues. I will be sharing this with CNV-docs, QE, and CNV-Devel very soon for updating: https://docs.google.com/document/d/1l1zoZ-ehsRawDyhWLHPTmfBpNS3J_ZBrtxZrvLpBFD4/edit

CNV team: please feel free to comment on this PR with requests, including links to the relevant Jiras/BZs that need release notes. Thanks!

Latest preview build: http://file.bos.redhat.com/bgaydos/013020/cnv/cnv-release-notes/cnv-release-notes.html

ousleyp commented:
@bgaydosrh I think we should add a note about the Virtual Machines dashboard in the "New and changed features" section under "Web console improvements." Maybe just something like:
`You can now view high-level information about your virtual machines by using the Virtual Machines dashboard in the OpenShift Container Platform web console. Access it by navigating to the *Workloads* -> *Virtual Machines* page.`

bgaydosrh commented:
Add a rel note for https://bugzilla.redhat.com/show_bug.cgi?id=1788645 and an associated doc change detailed therein.

bgaydosrh commented:
Add a rel note for https://bugzilla.redhat.com/show_bug.cgi?id=1754920 - Pick to 4.2 and 4.3

bgaydosrh commented:
Need a rel note for https://bugzilla.redhat.com/show_bug.cgi?id=1754920 Pick to 4.3 and 4.2

bgaydosrh commented:
> You can now view high-level information about your virtual machines by using the *Virtual Machines* dashboard in the OpenShift Container Platform web console. Access it by navigating to the *Workloads* -> *Virtual Machines* page and selecting a virtual machine.

Added the following to New and Changed Features:
* View high-level information about your virtual machines 
from the web console in the {product-title} dashboard. 
Access the information by navigating to the *Workloads* -> *Virtual Machines* page 
and selecting a virtual machine.

In the dashboard topics we refer to the {product-title} dashboard which I believe is within the web console, so correct me @ousleyp if that is not right.

bgaydosrh commented:
> Need a rel note for https://bugzilla.redhat.com/show_bug.cgi?id=1754920 Pick to 4.3 and 4.2

Added the following to Known Issues. See the BZ for more info.

When running virtctl image-upload to upload large VM disk images in qcow2 format, an end-of-file (EOF) error may be reported after the data is transmitted, even though the upload is either progressing normally or completed. Run the following command to check the status of an upload on a given PVC:

$ oc describe pvc <pvc-name> | grep cdi.kubevirt.io/storage.pod.phase

ousleyp commented:
> > You can now view high-level information about your virtual machines by using the _Virtual Machines_ dashboard in the OpenShift Container Platform web console. Access it by navigating to the _Workloads_ -> _Virtual Machines_ page and selecting a virtual machine.
> 
> Added the following to New and Changed Features:
> 
> * View high-level information about your virtual machines
>   from the web console in the {product-title} dashboard.
>   Access the information by navigating to the _Workloads_ -> _Virtual Machines_ page
>   and selecting a virtual machine.
> 
> In the dashboard topics we refer to the {product-title} dashboard which I believe is within the web console, so correct me @ousleyp if that is not right.

I don't think that's right, because `{product-title}` == OpenShift Container Platform. This dashboard is just one part of the web console, and it is just called Virtual Machines.

bgaydosrh commented:
> > > You can now view high-level information about your virtual machines by using the _Virtual Machines_ dashboard in the OpenShift Container Platform web console. Access it by navigating to the _Workloads_ -> _Virtual Machines_ page and selecting a virtual machine.
> > 
> > 
> > Added the following to New and Changed Features:
> > 
> > * View high-level information about your virtual machines
> >   from the web console in the {product-title} dashboard.
> >   Access the information by navigating to the _Workloads_ -> _Virtual Machines_ page
> >   and selecting a virtual machine.
> > 
> > In the dashboard topics we refer to the {product-title} dashboard which I believe is within the web console, so correct me @ousleyp if that is not right.
> 
> I don't think that's right, because `{product-title}` == OpenShift Container Platform. This dashboard is just one part of the web console, and it is just called Virtual Machines.

OK, I can change it back. Not sure what we should do with the dashboard topic except to make a note to change it to match the wording here.

dankenigsberg commented on cnv/cnv_release_notes/cnv-release-notes.adoc:
```diff
@@ -197,3 +161,27 @@ the nodes of any clusters where {CNVProductName} is deployed. The nodes might no
+----
+
+* When attempting to create a virtual machine on hardware with a Haswell CPU,
+the virtual machine creation fails and `libvert` declares
```
libvert->libvirt

bgaydosrh replied:
Done. Thanks!

dankenigsberg commented on cnv/cnv_release_notes/cnv-release-notes.adoc:
```diff
@@ -197,3 +161,27 @@ the nodes of any clusters where {CNVProductName} is deployed. The nodes might no
+command to fix the incorrect flag on the Haswell CPU:
++
+----
+$ oc label node $THE_NODE_NAME feature.node.kubernetes.io/cpu-model-Haswell=false --overwrite
```
@bgaydosrh can you double check with Karel Simon about this? Wouldn't the node labeler overwrite this on the next time it is running?

ksimon1 replied:
Node-labeller overwrites labels when it is restarted and after that there will be again  `cpu-model-Haswell=true`

bgaydosrh replied:
Dropping workaround per @ksimon1. cc: @dankenigsberg

bgaydosrh replied:
Just checking to verify if I can still say: "As a workaround, select a different CPU model, if possible." 

I know it's not great, but it's something. @ksimon1 @dankenigsberg

fabiand replied:
Yes. Let's keep the last recommendation.

bgaydosrh replied:
Done.

fabiand commented on cnv/cnv_release_notes/cnv-release-notes.adoc:
```diff
@@ -197,3 +161,27 @@ the nodes of any clusters where {CNVProductName} is deployed. The nodes might no
+the virtual machine creation fails and `libvert` declares
+`<model usable='yes'>Haswell</model>`. This is a change in behavior from
+previous verions of {CNVProductName}, where `libvirt` declared
+`<model usable='no'>Haswell</model>`.
```
I've proposed this text in the relevant bug, pasting it here (adjusted to the recent discussion):

```
"""
When attempting to create and kaunch a virtual machine using a Haswell CPU (model: Haswell), then the launch of the virtual machine fails. This is a bug due to incorrectly labeled nodes and a change in behaviour from previous versions of container-native virtualization, where VMs could be successfully launched on Haswell hosts.

As a workaround, select a different CPU model, if possible.
```

bgaydosrh replied:
Updated.

bgaydosrh commented on cnv/cnv-release-notes/cnv-release-notes.adoc:
```diff
@@ -16,52 +16,44 @@ include::modules/technology-preview.adoc[leveloffset=+2]
 
-=== Web console improvements
+* View high-level information about your virtual machines by
+using the *Virtual Machines* dashboard in the (product-name} web console.
```
Great catch, thanks. Those braces always trip me up and they are hard to see....

nellyc replied:
still incorrect.
where is it defined? (i assume it should be OCP and not CNV)

bgaydosrh replied:
Just fixed it. It is product-title that expands to OCP.

bgaydosrh commented on cnv/cnv-release-notes/cnv-release-notes.adoc:
```diff
@@ -16,52 +16,44 @@ include::modules/technology-preview.adoc[leveloffset=+2]
+* Some resources are improperly retained when removing {CNVProductName}. You
+must manually remove these resources in order to reinstall {CNVProductName}.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1712429[*BZ#1712429*]),
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1757705[*BZ#1757705*])
```
Removed link for 1757705 and moved the remaining 1712429 to Known Issues. As no workaround exists, that will have to do for the latter.

bgaydosrh commented on cnv/cnv-release-notes/cnv-release-notes.adoc:
```diff
@@ -72,8 +64,9 @@ obsolete IP address. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1686208[*
-* The virtual machines wizard does not load for users without administrator
-privileges. This issue is caused by missing permissions that allow users to load
+* Users without administrator privileges cannot add a network interface
+to a project in an L2 network using the virtual machine wiz[ard.
```
Removed, thanks. The weirdness of Asciidoc.....

bgaydosrh commented on cnv/cnv-release-notes/cnv-release-notes.adoc:
```diff
@@ -196,4 +160,32 @@ running either `oc adm drain` or `kubectl drain`. Do not run these commands on
+* When running `virtctl image-upload` to upload large VM disk images in `qcow2`
+format, an end-of-file (EOF) error may be reported after the data is
+transmitted, even though the upload is either progressing normally or completed.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1754920[*BZ#1754920*])
```
Thanks, replaced BZ number.

bgaydosrh commented on cnv/cnv-release-notes/cnv-release-notes.adoc:
```diff
@@ -196,4 +160,32 @@ running either `oc adm drain` or `kubectl drain`. Do not run these commands on
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1795182[*BZ#1795182*])
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1795186[*BZ#1795186*])
+
+* When running `virtctl image-upload` to upload large VM disk images in `qcow2`
```
OK, since we may need to get this merged sooner than later, we'll move ahead but we can always update later if we hear otherwise.... Assuming you'll follow up with Natalie, but if you want me to, let me know.

nellyc replied:
i believe Natalie marked it as verified

bgaydosrh replied:
I saw that. Thanks.

bgaydosrh commented on cnv/cnv-release-notes/cnv-release-notes.adoc:
```diff
@@ -196,4 +160,32 @@ running either `oc adm drain` or `kubectl drain`. Do not run these commands on
+$ oc describe pvc <pvc-name> | grep cdi.kubevirt.io/storage.pod.phase
+----
+
+* When attempting to create and launch a virtual machine using a Haswell CPU,
```
Fabian had looked at that and rejected it after he and I talked, but I have a ping out to him on hangouts to get his confirmation.

nellyc replied:
ack

bgaydosrh commented on modules/cnv-document-attributes.adoc:
```diff
@@ -26,4 +27,4 @@
 :Install_BookName: Installing container-native virtualization
 :Using_BookName: Using container-native virtualization
-:RN_BookName: Container-native virtualization 2.1 release notes
+:RN_BookName: Container-native virtualization release notes
```
No, we are moving the release number to the About heading. It makes more sense to keep the title of the doc generic.

nellyc replied:
ack

nellyc commented on cnv/cnv-release-notes/cnv-release-notes.adoc:
```diff
@@ -16,52 +16,42 @@ include::modules/technology-preview.adoc[leveloffset=+2]
 
 == Known issues
 
+* If you have {CNVProductName} 2.1.0 deployed, you must first upgrade to 2.2.0
```
do we want to link here to the 3 resolved BZs which were causing the issue?

bgaydosrh replied:
Waiting on feedback from others. PR is still being reviewed so we have some time.

ousleyp commented on cnv/cnv-release-notes/cnv-release-notes.adoc:
```diff
@@ -196,4 +162,32 @@ running either `oc adm drain` or `kubectl drain`. Do not run these commands on
+you can potentially exhaust the space on the partition, causing the node to
+be non-functional. Instead, create a separate partition
+and point the hostpath provisioner to that partition so it will not
+interfere with your Operating System.
```
s/Operating System/operating system

bgaydosrh replied:
Done.

bgaydosrh commented on cnv/cnv-release-notes/cnv-release-notes.adoc:
```diff
@@ -16,52 +16,42 @@ include::modules/technology-preview.adoc[leveloffset=+2]
-====
-
+* View high-level information about your virtual machines by
+using the *Virtual Machines* dashboard in the {product-name} web console.
```
Done.

bgaydosrh commented on cnv/cnv-release-notes/cnv-release-notes.adoc:
```diff
@@ -16,52 +16,42 @@ include::modules/technology-preview.adoc[leveloffset=+2]
+* When adding a disk to a virtual machine via the *Disks* tab in the web console,
+the added disk always has a `Filesystem` volumeMode, regardless of the volumeMode
+set in the `kubevirt-storage-class-default` ConfigMap.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1753688[*BZ#1753688*])
```
Done.

bgaydosrh commented on cnv/cnv-release-notes/cnv-release-notes.adoc:
```diff
@@ -16,52 +16,42 @@ include::modules/technology-preview.adoc[leveloffset=+2]
+
+* When navigating to the *Virtual Machines Console* tab, sometimes no content
+is displayed. As a workaround, use the serial console.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1753606[*BZ#1753606*])
```
Done.

bgaydosrh commented on cnv/cnv-release-notes/cnv-release-notes.adoc:
```diff
@@ -16,52 +16,42 @@ include::modules/technology-preview.adoc[leveloffset=+2]
-(link:https://bugzilla.redhat.com/show_bug.cgi?id=1708680[*BZ#1708680*])
+* When you attempt to list all instances of the {CNVProductName} operator from a
+browser, you receive a 404 (page not found) error.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1757526[*BZ#1757526*])
```
Done.

bgaydosrh commented on cnv/cnv-release-notes/cnv-release-notes.adoc:
```diff
@@ -16,52 +16,42 @@ include::modules/technology-preview.adoc[leveloffset=+2]
+workaround, remove the `CPUManager` entry from the `kubevirt-config` ConfigMap.
+Then, manually label the nodes with `cpumanager=true` before running virtual
+machines with guaranteed CPUs on your cluster.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1718944[*BZ#1718944*])
```
Done.

bgaydosrh commented on cnv/cnv-release-notes/cnv-release-notes.adoc:
```diff
@@ -16,52 +16,42 @@ include::modules/technology-preview.adoc[leveloffset=+2]
 == Known issues
 
+* If you have {CNVProductName} 2.1.0 deployed, you must first upgrade to 2.2.0
+before upgrading {product-title}.
```
There are but they are all from 2.1 and fixed. Nelly sent out a question to the OS group so I guess we'll see. In my opinion it would confuse the user.

bgaydosrh commented on cnv/cnv-release-notes/cnv-release-notes.adoc:
```diff
@@ -196,4 +162,32 @@ running either `oc adm drain` or `kubectl drain`. Do not run these commands on
+the launch of the virtual machine can fail. This is a bug due to incorrectly
+labeled nodes and a change in behavior from previous versions of
+container-native virtualization, where virtual machines could be successfully
+launched on Haswell hosts.
```
* When attempting to create and launch a virtual machine using a Haswell CPU,
the launch of the virtual machine can fail due to incorrectly labeled nodes.  
This is a change in behavior from previous versions of container-native 
virtualization, where virtual machines could be successfully launched on Haswell hosts. 
(link:https://bugzilla.redhat.com/show_bug.cgi?id=1781497[*BZ#1781497*])
+
As a workaround, select a different CPU model, if possible.

bgaydosrh commented on cnv/cnv-release-notes/cnv-release-notes.adoc:
```diff
@@ -196,4 +162,32 @@ running either `oc adm drain` or `kubectl drain`. Do not run these commands on
++
+As a workaround, select a different CPU model, if possible.
+
+* If you select a directory that shares space with your Operating System,
```
Done.
